veracruz-utils/src/runtime_manager_message.rs

```diff
@@ -35,68 +32,77 @@ pub enum VMStatus {
 // Command-and-control messages for the runtime manager.
 ///////////////////////////////////////////////////////////////////////////////
 
-/// An enumerated type describing messages passed between to/from the Runtime
-/// Manager enclave (These originate from the Untrusted Pass-through (Veracruz
-/// server)
+/// An enumerated type describing request messages passed to the Runtime
+/// Manager enclave (these originate from the Untrusted Pass-through (Veracruz
+/// server).
 /// These messages are inteded to be serialized using bincode before transport,
-/// and deserialized using bincode after transport
+/// and deserialized using bincode after transport.
 #[derive(Serialize, Deserialize, Debug)]
-pub enum RuntimeManagerMessage {
-    /// A message generated by an operation that did not return data, but did
-    /// return a status.  Most operations return data, but if they fail, they
-    /// will return a status set to `VMStatus::Fail` (or
-    /// `VMStatus::Unimplemented` if it is not implmeneted).  Parameters in
+pub enum RuntimeManagerRequest {
+    /// A request to start the attestation process
+    /// parameters:
+    /// Vec<u8> - the challenge value
+    /// i32     - the challenge ID
+    Attestation(Vec<u8>, i32),
+    /// A request to close an already established TLS session.  Parameters in
     /// order are:
-    /// - The status.
-    Status(VMStatus),
-    #[cfg(any(feature = "nitro", feature = "linux"))]
+    /// - The Session ID of the session to be closed.
+    CloseTlsSession(u32),
+    /// Request TLS data from the enclave.  Parameters in order are:
+    /// - The Session ID of the TLS session to request data from.
+    GetTlsData(u32),
+    /// Request to determine if the TLS Session needs data to be sent to it.
+    /// Parameters in order are:
+    /// - The Session ID of the TLS session.
+    GetTlsDataNeeded(u32),
```

Comment:
Just musing again.
The semantics of this one make even less sense (and it's always been a problem):
"Request Get TLS Data Needed"
"Request To learn if TLS Data is needed"

Nothing seems to work. Nevermind me.

Reply:
Not really. But I don't think there's a good solution.